src/components/JaegerIntegration/JaegerResults/SpanTableItem.tsx

```diff
@@ -33,47 +25,12 @@ const linkStyle = style({
   fontSize: 14
 });
 
-export type SpanItemData = Span & {
-  type: 'envoy' | 'http' | 'tcp' | 'unknown';
-  component: string;
-  namespace: string;
-  app: string;
-  linkToApp: string;
-  workload?: string;
-  pod?: string;
-  linkToWorkload?: string;
-  info: OpenTracingBaseInfo;
-};
-
-// Extracts some information from a span to make it suitable for table-display
-export const buildItemData = (span: Span, defaultNamespace: string): SpanItemData => {
-  const { type, info } = extractSpanInfo(span);
-  const workloadNs = getWorkloadFromSpan(span);
-  const split = span.process.serviceName.split('.');
-  const app = split[0];
-  const namespace = workloadNs ? workloadNs.namespace : split.length > 1 ? split[1] : defaultNamespace;
-  const linkToApp = '/namespaces/' + namespace + '/applications/' + app;
-  const linkToWorkload = workloadNs ? '/namespaces/' + namespace + '/workloads/' + workloadNs.workload : undefined;
-  return {
-    ...span,
-    type: type,
-    info: info,
-    component: info.component || 'unknown',
-    namespace: namespace,
-    app: app,
-    linkToApp: linkToApp,
-    workload: workloadNs?.workload,
-    pod: workloadNs?.pod,
-    linkToWorkload: linkToWorkload
-  };
-};
-
-type RowProps = SpanItemData & {
+type RowProps = RichSpanData & {
   toggledLinks?: string;
   setToggledLinks: (key: string) => void;
   externalURL?: string;
   onClickFetchStats: () => void;
-  metricsStats: { [key: string]: MetricsStats };
+  metricsStats: Map<string, MetricsStats>;
```

Comment:
I think this Map doesn't come from an api call, it's built internally

Reply:
Yes that's correct, is that a problem?